### ISSUE TITLE
how could i use pytorhch3d v0.2.5 to render face 

### ISSUE THREAD

**Issue by ZzZzEnG** [facebookresearch/pytorch3d#422]
## ❓ Questions on how to use PyTorch3D

         because you change the textures and my previous code could not work ,and i do not know how to fix my code to render the  face  can you give the example ?
   
![2020-11-03 17-53-07屏幕截图](https://user-images.githubusercontent.com/32928342/97971130-89288e80-1dfd-11eb-9a12-38f475c20d29.png)

NOTE: Please look at the existing list of Issues tagged with the label ['question`](https://github.com/facebookresearch/pytorch3d/issues?q=label%3Aquestion) or ['how-to`](https://github.com/facebookresearch/pytorch3d/issues?q=label%3A%22how+to%22). **Only open a new issue if you cannot find an answer there**.

Also note the following:

1. If you encountered any errors or unexpected issues while using PyTorch3D and need help resolving them,
   please use the "Bugs / Unexpected behaviors" issue template.

2. We do not answer general machine learning / computer vision questions that are not specific to
	 PyTorch3D, such as how a model works or what algorithm/methods can be
	 used to achieve X.

**Comment by bottler** [comment]
As mentioned in the "breaking changes" section of the release note, `Textures` has moved from `pytorch3d.structures` to `pytorch3d.renderer`. So take it out from your first import and put it in the big list in your second import. (Note: it's easier if you can share your code in text form, e.g. then I could easily demonstrate what I mean.)

**Comment by ZzZzEnG** [comment]
> As mentioned in the "breaking changes" section of the [release note](https://github.com/facebookresearch/pytorch3d/releases/tag/v0.2.5), `Textures` has moved from `pytorch3d.structures` to `pytorch3d.renderer`. So take it out from your first import and put it in the big list in your second import. (Note: it's easier if you can share your code in text form, e.g. then I could easily demonstrate what I mean.)

here is the code that can be used in the previous version of the pytorch3d ,i use it to render the face ,but now it does not work  because of the Textures has changed ,can you help me to fix it 
def render_img(face_shape, face_color, facemodel, image_size=224, fx=1015.0, fy=1015.0, px=112.0, py=112.0, device='cuda:0'):
    '''
        ref: https://github.com/facebookresearch/pytorch3d/issues/184
        The rendering function (just for test)
        Input:
            face_shape:  Tensor[1, 35709, 3]
            face_color: Tensor[1, 35709, 3] in [0, 1]
            facemodel: contains `tri` (triangles[70789, 3], index start from 1)
    '''
    from pytorch3d.structures import Meshes, Textures
    from pytorch3d.renderer import (
        SfMPerspectiveCameras,
        PointLights,
        RasterizationSettings,
        MeshRenderer,
        MeshRasterizer,
        SoftPhongShader,
        BlendParams
    )

    face_color = Textures(verts_rgb=face_color.to(device))
    face_buf = torch.from_numpy(facemodel.tri - 1)  # index start from 1
    face_idx = face_buf.unsqueeze(0)

    mesh = Meshes(face_shape.to(device), face_idx.to(device), face_color)

    R = torch.eye(3).view(1, 3, 3).to(device)
    R[0, 0, 0] *= -1.0
    T = torch.zeros([1, 3]).to(device)

    half_size = (image_size - 1.0) / 2
    focal_length = torch.tensor([fx / half_size, fy / half_size], dtype=torch.float32).reshape(1, 2).to(device)
    principal_point = torch.tensor([(half_size - px) / half_size, (py - half_size) / half_size], dtype=torch.float32).reshape(1, 2).to(device)

    cameras = SfMPerspectiveCameras(
        device=device,
        R=R,
        T=T,
        focal_length=focal_length,
        principal_point=principal_point
    )

    raster_settings = RasterizationSettings(
        image_size=image_size,
        blur_radius=0.0,
        faces_per_pixel=1
    )

    lights = PointLights(
        device=device,
        ambient_color=((1.0, 1.0, 1.0),),
        diffuse_color=((0.0, 0.0, 0.0),),
        specular_color=((0.0, 0.0, 0.0),),
        location=((0.0, 0.0, 1e5),)
    )

    blend_params = BlendParams(background_color=(0.0, 0.0, 0.0))

    renderer = MeshRenderer(
        rasterizer=MeshRasterizer(
            cameras=cameras,
            raster_settings=raster_settings
        ),
        shader=SoftPhongShader(
            device=device,
            cameras=cameras,
            lights=lights,
            blend_params=blend_params
        )
    )
    images = renderer(mesh)
    images = torch.clamp(images, 0.0, 1.0)
    return images

**Comment by ZzZzEnG** [comment]
> > As mentioned in the "breaking changes" section of the [release note](https://github.com/facebookresearch/pytorch3d/releases/tag/v0.2.5), `Textures` has moved from `pytorch3d.structures` to `pytorch3d.renderer`. So take it out from your first import and put it in the big list in your second import. (Note: it's easier if you can share your code in text form, e.g. then I could easily demonstrate what I mean.)
> 
> here is the code that can be used in the previous version of the pytorch3d ,i use it to render the face ,but now it does not work because of the Textures has changed ,can you help me to fix it
> 
> def render_img(face_shape, face_color, facemodel, image_size=224, fx=1015.0, fy=1015.0, px=112.0, py=112.0,device='cuda:0'):
> '''
> ref: #184
> The rendering function (just for test)
> Input:
> face_shape: Tensor[1, 35709, 3]
> face_color: Tensor[1, 35709, 3] in [0, 1]
> facemodel: contains `tri` (triangles[70789, 3], index start from 1)
> '''
> from pytorch3d.structures import Meshes
> from pytorch3d.renderer import TexturesUV, TexturesVertex, TexturesAtlas
> from pytorch3d.renderer import (
> SfMPerspectiveCameras,
> PointLights,
> RasterizationSettings,
> MeshRenderer,
> MeshRasterizer,
> SoftPhongShader,
> BlendParams
> )
> 
> ```
> face_color = TexturesUV(face_color.to(device))##xiugai
> face_buf = torch.from_numpy(facemodel.tri - 1)  # index start from 1
> face_idx = face_buf.unsqueeze(0)
> 
> mesh = Meshes(face_shape.to(device), face_idx.to(device), face_color)
> 
> R = torch.eye(3).view(1, 3, 3).to(device)
> R[0, 0, 0] *= -1.0
> T = torch.zeros([1, 3]).to(device)
> 
> half_size = (image_size - 1.0) / 2
> focal_length = torch.tensor([fx / half_size, fy / half_size], dtype=torch.float32).reshape(1, 2).to(device)
> principal_point = torch.tensor([(half_size - px) / half_size, (py - half_size) / half_size], dtype=torch.float32).reshape(1, 2).to(device)
> 
> cameras = SfMPerspectiveCameras(
>     device=device,
>     R=R,
>     T=T,
>     focal_length=focal_length,
>     principal_point=principal_point
> )
> 
> raster_settings = RasterizationSettings(
>     image_size=image_size,
>     blur_radius=0.0,
>     faces_per_pixel=1
> )
> 
> lights = PointLights(
>     device=device,
>     ambient_color=((1.0, 1.0, 1.0),),
>     diffuse_color=((0.0, 0.0, 0.0),),
>     specular_color=((0.0, 0.0, 0.0),),
>     location=((0.0, 0.0, 1e5),)
> )
> 
> blend_params = BlendParams(background_color=(0.0, 0.0, 0.0))
> 
> renderer = MeshRenderer(
>     rasterizer=MeshRasterizer(
>         cameras=cameras,
>         raster_settings=raster_settings
>     ),
>     shader=SoftPhongShader(
>         device=device,
>         cameras=cameras,
>         lights=lights,
>         blend_params=blend_params
>     )
> )
> images = renderer(mesh)
> images = torch.clamp(images, 0.0, 1.0)
> return images
> ```

def render_img(face_shape, face_color, facemodel, image_size=224, fx=1015.0, fy=1015.0, px=112.0, py=112.0, device='cuda:0'):
    '''
        ref: https://github.com/facebookresearch/pytorch3d/issues/184
        The rendering function (just for test)
        Input:
            face_shape:  Tensor[1, 35709, 3]
            face_color: Tensor[1, 35709, 3] in [0, 1]
            facemodel: contains `tri` (triangles[70789, 3], index start from 1)
    '''
    from pytorch3d.structures import Meshes
    from pytorch3d.renderer import TexturesUV, TexturesVertex, TexturesAtlas
    from pytorch3d.renderer import (
        SfMPerspectiveCameras,
        PointLights,
        RasterizationSettings,
        MeshRenderer,
        MeshRasterizer,
        SoftPhongShader,
        BlendParams
    )

    face_color = TexturesUV(face_color.to(device))##xiugai
    face_buf = torch.from_numpy(facemodel.tri - 1)  # index start from 1
    face_idx = face_buf.unsqueeze(0)

    mesh = Meshes(face_shape.to(device), face_idx.to(device), face_color)

    R = torch.eye(3).view(1, 3, 3).to(device)
    R[0, 0, 0] *= -1.0
    T = torch.zeros([1, 3]).to(device)

    half_size = (image_size - 1.0) / 2
    focal_length = torch.tensor([fx / half_size, fy / half_size], dtype=torch.float32).reshape(1, 2).to(device)
    principal_point = torch.tensor([(half_size - px) / half_size, (py - half_size) / half_size], dtype=torch.float32).reshape(1, 2).to(device)

    cameras = SfMPerspectiveCameras(
        device=device,
        R=R,
        T=T,
        focal_length=focal_length,
        principal_point=principal_point
    )

    raster_settings = RasterizationSettings(
        image_size=image_size,
        blur_radius=0.0,
        faces_per_pixel=1
    )

    lights = PointLights(
        device=device,
        ambient_color=((1.0, 1.0, 1.0),),
        diffuse_color=((0.0, 0.0, 0.0),),
        specular_color=((0.0, 0.0, 0.0),),
        location=((0.0, 0.0, 1e5),)
    )

    blend_params = BlendParams(background_color=(0.0, 0.0, 0.0))

    renderer = MeshRenderer(
        rasterizer=MeshRasterizer(
            cameras=cameras,
            raster_settings=raster_settings
        ),
        shader=SoftPhongShader(
            device=device,
            cameras=cameras,
            lights=lights,
            blend_params=blend_params
        )
    )
    images = renderer(mesh)
    images = torch.clamp(images, 0.0, 1.0)
    return images

**Comment by ZzZzEnG** [comment]
> As mentioned in the "breaking changes" section of the [release note](https://github.com/facebookresearch/pytorch3d/releases/tag/v0.2.5), `Textures` has moved from `pytorch3d.structures` to `pytorch3d.renderer`. So take it out from your first import and put it in the big list in your second import. (Note: it's easier if you can share your code in text form, e.g. then I could easily demonstrate what I mean.)

             and you can also find the code at this place   at the line 275,the  function render_image
               https://github.com/yuhaoooo/Deep3DFaceModel-Pytorch/blob/master/reconstruction_mesh.py

**Comment by ZzZzEnG** [comment]
> As mentioned in the "breaking changes" section of the [release note](https://github.com/facebookresearch/pytorch3d/releases/tag/v0.2.5), `Textures` has moved from `pytorch3d.structures` to `pytorch3d.renderer`. So take it out from your first import and put it in the big list in your second import. (Note: it's easier if you can share your code in text form, e.g. then I could easily demonstrate what I mean.)

my problem has solved thank you very much
